staging/src/k8s.io/apiserver/pkg/apis/apiserver/validation/validation_test.go

```diff
@@ -557,6 +562,140 @@ func TestValidateAuthorizationConfiguration(t *testing.T) {
 			knownTypes:      sets.NewString(string("Webhook")),
 			repeatableTypes: sets.NewString(string("Webhook")),
 		},
+		{
+			name: "bare minimum configuration with Webhook and MatchConditions",
+			configuration: api.AuthorizationConfiguration{
+				Authorizers: []api.AuthorizerConfiguration{
+					{
+						Type: "Webhook",
+						Name: "default",
+						Webhook: &api.WebhookConfiguration{
+							Timeout:                                  metav1.Duration{Duration: 5 * time.Second},
+							AuthorizedTTL:                            metav1.Duration{Duration: 5 * time.Minute},
+							UnauthorizedTTL:                          metav1.Duration{Duration: 30 * time.Second},
+							FailurePolicy:                            "NoOpinion",
+							SubjectAccessReviewVersion:               "v1",
+							MatchConditionSubjectAccessReviewVersion: "v1",
+							ConnectionInfo: api.WebhookConnectionInfo{
+								Type: "InClusterConfig",
+							},
+							MatchConditions: []api.WebhookMatchCondition{
+								{
+									Expression: "has(request.resourceAttributes) && request.resourceAttributes.namespace == 'kube-system'",
+								},
+								{
+									Expression: "request.user == 'admin'",
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedErrList: field.ErrorList{},
```

Comment:
`expectedErrList` predates this PR. It's pretty useful when you want to make sure there are multiple errors returned especially because `ValidateAuthorizationConfiguration` returns a list
e.g. where multiple errors are checked: https://github.com/kubernetes/kubernetes/blob/ba2c8a4a912aa5a83f663e733f7f1f49369fc757/staging/src/k8s.io/apiserver/pkg/apis/apiserver/validation/validation_test.go#L628-L629

and each error in the list does get validated against the expected values: 
https://github.com/kubernetes/kubernetes/blob/ba2c8a4a912aa5a83f663e733f7f1f49369fc757/staging/src/k8s.io/apiserver/pkg/apis/apiserver/validation/validation_test.go#L1315

WDYT?

Reply:
The goal is to validate the actual error message the user will see. With the `expectedErr` string you can compare the `fields.Aggregate()` error message the user will see: for example `[issuer.claimMappings.username.expression: Invalid value: "foo.bar": expression is not supported when StructuredAuthenticationConfiguration feature gate is disabled, issuer.claimMappings.username.expression: Invalid value: "foo.bar": compilation failed: ERROR: <input>:1:1: undeclared reference to 'foo' (in container '')`.

By checking the actual error message we expect, we'll ensure we aren't logging something that's not decipherable.

Reply:
I think it's fine to do this in a follow-up. Not a blocker for this PR.

Reply:
updated